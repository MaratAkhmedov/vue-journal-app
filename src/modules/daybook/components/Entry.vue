<template>
    <div class="entry-container mb-3 pointer p-2">

        <!-- Title -->
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">15</span>
            <span class="mx-1 fs-5">Julio</span>
            <span class="mx-2 fw-light">2021, jueves</span>
        </div>

        <div class="entry-description">
            Minim anim ullamco exercitation aliquip consectetur duis veniam labore magna consequat cillum excepteur mollit. Sit consectetur adipisicing qui velit Lorem exercitation consectetur ea eu dolore fugiat. Labore elit esse non Lorem officia velit nisi. Officia pariatur ut labore consequat. Reprehenderit aliquip voluptate deserunt incididunt aliquip.
        </div>
    </div>
</template>
<script>
export default {

}
</script>
<style lang="scss">
    .entry-container {
        border-bottom: 1px solid #2c3e50;
        transition: 0.2s all ease-in;

        &:hover {
            background-color: lighten($color: grey, $amount: 45);
            transition: 0.2s all ease-in;
        }
    }

    .entry-description{
        font-size: 12px;
    }
</style>