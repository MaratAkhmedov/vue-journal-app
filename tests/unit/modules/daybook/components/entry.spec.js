import { shallowMount } from "@vue/test-utils";

import Entry from "@/modules/daybook/components/Entry.vue";
import journalState from "../../../mock-data/test-journal-state";

describe('Entry component tests', () => {
    let wrapper, mockRouter, defaultMockEntry

    beforeEach(() => {
        mockRouter = {
            push: jest.fn()
        }
        defaultMockEntry = journalState.entries[0]
        
        wrapper = shallowMount(Entry, {
            props: {
                entry: defaultMockEntry
            },
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })
    })

    it('should match with a snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    });

    it('should redirect on click on .entry-container', () => {
        wrapper.find('.entry-container').trigger('click')
        expect(mockRouter.push).toHaveBeenCalledWith({name: 'daybook-entry', params: {id: defaultMockEntry.id}})
    });

    it('check shortText computed property', async() => {
        const vm = wrapper.vm
        //show full text when it is short
        expect(vm.shortText.length).toBe(defaultMockEntry.text.length)

        const mockWithLargeText = {...defaultMockEntry}
        mockWithLargeText.text = "Reprehenderit elit elit proident tempor laboris in officia irure dolore nostrud. Consequat sunt dolor aute eu non in minim non ut cupidatat. Do fugiat cupidatat laborum consectetur laborum qui amet.Quis eu aute incididunt elit sit tempor. Do qui id in elit aliquip irure cillum dolor eiusmod officia cupidatat proident cillum magna. Cupidatat eu tempor elit magna est non pariatur occaecat minim sit.Magna eu amet elit ipsum sit aliquip et laborum sunt duis amet. In quis sit dolor sit Lorem fugiat id eiusmod officia consectetur. Veniam cupidatat cillum sit aliqua officia ea excepteur fugiat nisi ullamco minim fugiat. Ad voluptate nulla laboris ut excepteur aliqua. Aute fugiat incididunt incididunt eu laborum pariatur deserunt dolore consequat velit ut dolore officia."
        await wrapper.setProps({entry: mockWithLargeText})

        expect(vm.shortText.slice(0, -3).length).toBe(150)
    });
});