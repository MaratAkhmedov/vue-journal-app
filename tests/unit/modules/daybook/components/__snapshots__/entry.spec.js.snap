// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component tests should match with a snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Title -->
  <div class="entry-title d-flex">
    <entry-date-stub datestring="Sun Aug 20 2023"></entry-date-stub>
  </div>
  <div class="entry-description">
    <p>Hola mundo mock 1</p>
  </div>
</div>
`;
